### PR TITLE
Use systemd if found instead of testing for os version

### PR DESCRIPTION
**Pull request by juergenthomann**

### CODE REVIEW

File: src/check_smartmon.py [hunk 1]
```diff
@@ -106,43 +154,45 @@
...
+    CSV_PATH = conf_data['CSV_PATH']
     HDD_NAMES = conf_data['HDD_NAMES']
-    ALERTS    = conf_data['ALERTS']
+    ALERTS = conf_data['ALERTS']
 except IOError as e:
     pass
 
-os_major = platform.dist()[1].split('.')[0]
-if os_major == "8":
+if os.path.exists('/bin/systemctl'):
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Is this really the only location where `systemctl` could be? If there are more possibilities, maybe another solution could be to try to run the command and check the exit status?

File: src/check_smartmon.py [hunk 2]
```diff
@@ -160,12 +210,18 @@
         smart_m = int(smart_date.split(":")[1])
         smart_s = int(smart_date.split(":")[2])
 
-        future = datetime.datetime.now() - datetime.datetime(smart_year, smart_mon, smart_day, smart_h, smart_m, smart_s)
-        smart_date_in_seconds = int(future.days * 24 * 60 * 60 + future.seconds)
+        future = datetime.datetime.now() - datetime.datetime(smart_year,
+                                                             smart_mon,
+                                                             smart_day,
+                                                             smart_h, smart_m,
+                                                             smart_s)
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: That's ugly, rather:
``` python
from datetime import datetime

future = datetime.now() - datetime(
    smart_year, smart_mon, smart_day, smart_h, smart_m, smart_s
)
```

Or maybe also store the second datetime in an own variable.

File: src/check_smartmon.py [hunk 3]
```diff
@@ -160,12 +210,18 @@
...
+        smart_date_in_seconds = int(
+            future.days * 24 * 60 * 60 + future.seconds)
 
         # Throw error if csv data is more than 6 hours old!
         if smart_date_in_seconds > 6 * 60 * 60:
-            senddata.append("SMART-CSV-Data is > %s hours old! (%s%s)\n" % (str(smart_date_in_seconds/3600), CSV_PATH, csvfile))
+            senddata.append("SMART-CSV-Data is > %s hours old! (%s%s)\n" % (
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Please use `format` method.

File: src/check_smartmon.py [hunk 4]
```diff
@@ -175,45 +231,63 @@
...
                 else:
                     smart_value = int(element[2])
             else:
                 smart_value = int(element[1])
 
             if not int(smart_id) in ALERTS[manufacturer]:
-                senddata.append("Error: No id %s in %s-config found. Please configure this value.\n" % (smart_id, manufacturer))
+                senddata.append(
+                    "Error: No id %s in %s-config found. "
+                    "Please configure this value.\n" % (
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Please use `format` method.

File: src/check_smartmon.py [hunk 5]
```diff
@@ -175,45 +231,63 @@
...
+            if not ('or' in ALERTS[manufacturer][smart_id] and smart_value ==
+                    ALERTS[manufacturer][smart_id]['or']):
                 if smart_value < ALERTS[manufacturer][smart_id]['min']:
-                    senddata.append( "%s: [id=%s] '%s' is '%s' which is lower than '%s'\n" % ( 
-                        manufacturer, smart_id, ALERTS[manufacturer][smart_id]['description'], smart_value, ALERTS[manufacturer][smart_id]['min']))
-                    if exit_status < ALERTS[manufacturer][smart_id]['exit_status']:
-                        exit_status = ALERTS[manufacturer][smart_id]['exit_status']
+                    senddata.append(
+                        "%s: [id=%s] '%s' is '%s' which is lower than "
+                        "'%s'\n" % (
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Please use `format` method.

File: src/check_smartmon.py [hunk 6]
```diff
@@ -175,45 +231,63 @@
...
+                            manufacturer, smart_id,
+                            ALERTS[manufacturer][smart_id]['description'],
+                            smart_value,
+                            ALERTS[manufacturer][smart_id]['min']))
+                    if exit_status < \
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Please use braces to cover multiline python statements.

File: src/check_smartmon.py [hunk 7]
```diff
@@ -175,45 +231,63 @@
...
+                            'exit_status']
 
                 if smart_value > ALERTS[manufacturer][smart_id]['max']:
-                    senddata.append( "%s: [id=%s] '%s' is '%s' which is higher than '%s\n" % (
-                        manufacturer, smart_id, ALERTS[manufacturer][smart_id]['description'], smart_value, ALERTS[manufacturer][smart_id]['max']))
-                    if exit_status < ALERTS[manufacturer][smart_id]['exit_status']:
-                        exit_status = ALERTS[manufacturer][smart_id]['exit_status']
+                    senddata.append(
+                        "%s: [id=%s] '%s' is '%s' which is higher than "
+                        "'%s\n" % (
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Please use `format` method.

File: src/check_smartmon.py [hunk 8]
```diff
@@ -175,45 +231,63 @@
...
+                            manufacturer, smart_id,
+                            ALERTS[manufacturer][smart_id]['description'],
+                            smart_value,
+                            ALERTS[manufacturer][smart_id]['max']))
+                    if exit_status < \
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Please use braces to cover multiline python statements.

File: src/check_smartmon.py [hunk 9]
```diff
@@ -175,45 +231,63 @@
...
+if 0 < kingston_gb_written < 99:
     for i in range(len(senddata)):
         if 'KINGSTON: [id=13]' in senddata[i]:
             del senddata[i]
     if len(senddata) == 0:
         exit_status = 0
 
 if exit_status == 1:
-    print "WARNING: " + " ".join(senddata)
+    print("WARNING: " + " ".join(senddata))
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Please use `format` method.

File: src/check_smartmon.py [hunk 10]
```diff
@@ -175,45 +231,63 @@
...
 elif exit_status == 2:
-    print "CRITICAL: " + " ".join(senddata)
+    print("CRITICAL: " + " ".join(senddata))
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Please use `format` method.

File: src/check_smartmon.py [hunk 11]
```diff
@@ -160,12 +210,18 @@
...
 
-        future = datetime.datetime.now() - datetime.datetime(smart_year, smart_mon, smart_day, smart_h, smart_m, smart_s)
-        smart_date_in_seconds = int(future.days * 24 * 60 * 60 + future.seconds)
+        future = datetime.datetime.now() - datetime.datetime(smart_year,
+                                                             smart_mon,
+                                                             smart_day,
+                                                             smart_h, smart_m,
+                                                             smart_s)
+        smart_date_in_seconds = int(
+            future.days * 24 * 60 * 60 + future.seconds)
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: `24 * 60 * 60` could be stored in an own variable also as constant.

File: src/check_smartmon.py [hunk 12]
```diff
@@ -35,69 +37,115 @@
 #   /etc/nagios-plugins/check_smartmon.cfg
 # on each server (distributed by puppet)!
 
-### DEFAULT CONFIG VALUES ### (if /etc/nagios-plugins/check_smartmon.cfg doesn't exist!)
+### DEFAULT CONFIG VALUES ###
+# if /etc/nagios-plugins/check_smartmon.cfg doesn't exist!
 #
 # CSV_PATH: This value should never change
-CSV_PATH="/var/lib/smartmontools/"
+CSV_PATH = "/var/lib/smartmontools/"
```
Comment (TheFRedFox) [inline src/check_smartmon.py]: Single-quote.

File: src/check_smartmon.py (hunk 1 above)
Comment (juergenthomann) [reply src/check_smartmon.py]: At least on Debian like systems it is the only place. To support other systems we have to check for a different unit name as well.

Comment (TheFRedFox) [reply src/check_smartmon.py]: Ok, guess when there are more dependencies than just the path, then it is fair enough to keep it like that.

**Comment by TuxPowered42** [comment]
It would be way better to not duplicate functionality of other checks. On systems with systemd we have check_systemd_units which will verify if *all* daemons are running. On older systems it is enough to check csv files are not fresh which should be good enough. And we hopefully don't have such systems anymore.

**Pull request closed without merging**